web/components/templates/rateLimit/RateLimitRulesView.tsx

```diff
@@ -0,0 +1,380 @@
+import { P } from "@/components/ui/typography";
+import { Col } from "../../layout/common/col";
+import { useState } from "react";
+import RateLimitRuleModal from "./RateLimitRuleModal";
+import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+import { useOrg } from "@/components/layout/org/organizationContext";
+import { $JAWN_API } from "@/lib/clients/jawn";
+import { components } from "@/lib/clients/jawnTypes/private";
+import useNotification from "@/components/shared/notification/useNotification";
+import {
+  Table,
+  TableHeader,
+  TableRow,
+  TableHead,
+  TableBody,
+  TableCell,
+} from "@/components/ui/table";
+import { Button } from "@/components/ui/button";
+import { PiPlusBold } from "react-icons/pi";
+import { Badge } from "@/components/ui/badge";
+import { Trash2, Pencil } from "lucide-react";
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from "@/components/ui/alert-dialog";
+import LoadingAnimation from "@/components/shared/loadingAnimation";
+
+type RateLimitRuleView = components["schemas"]["RateLimitRuleView"];
+type UpdateRateLimitRuleParams =
+  components["schemas"]["UpdateRateLimitRuleParams"];
+
+const RateLimitRulesView = () => {
+  const org = useOrg();
+  const [isCreateModalOpen, setIsCreateModalOpen] = useState(false);
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const [editingRule, setEditingRule] = useState<RateLimitRuleView | null>(
+    null
+  );
+  const { setNotification } = useNotification();
+  const queryClient = useQueryClient();
+
+  const { data, isLoading, error, isError } = useQuery({
+    queryKey: ["rateLimits", org?.currentOrg?.id],
+    queryFn: async () => {
+      if (!org?.currentOrg?.id) {
+        return null; // Or throw an error, depending on desired behavior
+      }
+      const response = await $JAWN_API.GET("/v1/rate-limits");
+      return response;
+    },
+    enabled: !!org?.currentOrg?.id,
+  });
```

Comment:
Not a big deal, but just want to show you that you can get the same thing done with way less code....

```suggestion
  const { data, isLoading, error, isError } = $JAWN_API.useQuery("get", "/v1/rate-limits", {}, {
    enabled: !!org?.currentOrg?.id,
  });
```